sam/bot.py

```diff
@@ -236,15 +236,22 @@ async def add_message(
                 purpose="assistants",
             )
             file_ids.append(new_file.id)
-    await client.beta.threads.messages.create(
-        thread_id=thread_id,
-        content=content,
-        role=Roles.USER,
-        attachments=[
-            {"file_id": file_id, "tools": [{"type": "file_search"}]}
-            for file_id in file_ids
-        ],
-    )
+
+    try:
+        await client.beta.threads.messages.create(
+            thread_id=thread_id,
+            content=content,
+            role=Roles.USER,
+            attachments=[
+                {"file_id": file_id, "tools": [{"type": "file_search"}]}
+                for file_id in file_ids
+            ],
+        )
+    except openai.BadRequestError as e:
+        error = OSError()
+        error.strerror = e.body["message"]
+        raise error
```

Comment:
You are swallowing the original stack trace.
```suggestion
    except openai.BadRequestError as e:
        raise OSError("A proper message") from e
```

It's also a bit strange to capture a more specific error and to replace it with a less descriptive one.

Reply:
I'll give this a shot. My overall goal was to pass the most useful bit of information back to the user.